test/integration/run-command/run_cursor_command.test.ts

```diff
@@ -0,0 +1,53 @@
+import { expect } from 'chai';
+
+import { CommandStartedEvent, Db, MongoClient } from '../../mongodb';
+
+describe('class RunCommandCursor', () => {
+  let client: MongoClient;
+  let db: Db;
+  let commandsStarted: CommandStartedEvent[];
+
+  beforeEach(async function () {
+    client = this.configuration.newClient({}, { monitorCommands: true });
+    db = client.db();
+    await db.dropDatabase().catch(() => null);
+    await db
+      .collection<{ _id: number }>('collection')
+      .insertMany([{ _id: 0 }, { _id: 1 }, { _id: 2 }]);
+    commandsStarted = [];
+    client.on('commandStarted', started => commandsStarted.push(started));
+  });
+
+  afterEach(async function () {
+    commandsStarted = [];
+    await client.close();
+  });
+
+  it('should only run init command once', async () => {
```

Comment:
I don't really understand the structure of this test. Can you break it apart and/or clarify the test case title so that it's clear what conditions you are testing? Feel free to ping me on slack to discuss this further.

Reply:
This is general to our abstract cursor, but also it may not be a test we need to keep. taking this offline

Reply:
Refactored test to be clear about the interleaving of iterators